homeassistant/components/yamaha/media_player.py

```diff
@@ -193,6 +193,8 @@ class YamahaDevice(MediaPlayerEntity):
     def __init__(self, name, receiver, source_ignore, source_names, zone_names):
         """Initialize the Yamaha Receiver."""
         self.receiver = receiver
+        if name == DEFAULT_NAME and not receiver.model_name is None and not receiver.model_name == "Unknown":
+            name = 'Yamaha %s' % receiver.model_name
```

Comment:
Changing the default name is a breaking change as this integration doesn't set a unique id for the entities. I don't think we should do that.

The better approach is to modernize the integration to use config entries, device info and set entity unique id and follow our entity naming guidelines:

https://developers.home-assistant.io/docs/core/entity#entity-naming

https://developers.home-assistant.io/docs/config_entries_config_flow_handler

This is a bigger task but will result in a much better user experience.

Reply:
Yeah, I agree with that. This is breaking, which is unneeded as there is a non-breaking path available.

Going to friendly decline and close this PR for that reason. If you want to start on building any of the above @sredna, that would be greatly appreciated 🙏 

../Frenck

Reply:
I'm actually planning to set the unique id in another PR. I suppose you want both in the same PR?

As it currently stands, even if I assign a specific name in my yaml config, I still end up with multiple Yamaha entities sometimes (not static IP).

Reply:
The entity unique id feature needs to be released first before we can change default names without causing the entity_id to change, which is a breaking change.